From 4a5f09015183bb1744e39116837243bb86784607 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 26 Nov 2015 16:30:08 +0800
Subject: [PATCH 0692/5242] MLK-11911-4 mxc IPUv3: pixel clk: Free di_div clk
 memory if clk_regiser fails

commit  8c82dc4ada98d778715b332b8b03a467b2726276 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch fixes the following issue reported by Coverity:
	if (IS_ERR(clk))
freed_arg: kfree frees clk. [Note: The source code implementation of the
function has been overridden by a builtin model.]
		kfree(clk);

Use after free (USE_AFTER_FREE)
use_after_free: Using freed pointer clk.
	return clk;

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/ipu3/ipu_pixel_clk.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/ipu3/ipu_pixel_clk.c b/drivers/mxc/ipu3/ipu_pixel_clk.c
index 4efb6b7..7d3e96c 100644
--- a/drivers/mxc/ipu3/ipu_pixel_clk.c
+++ b/drivers/mxc/ipu3/ipu_pixel_clk.c
@@ -238,7 +238,7 @@ struct clk *clk_register_div_pix_clk(struct device *dev, const char *name,
 
 	clk = clk_register(dev, &di_div->hw);
 	if (IS_ERR(clk))
-		kfree(clk);
+		kfree(di_div);
 
 	return clk;
 }
-- 
1.7.9.5

